src/main/java/com/google/devtools/build/lib/bazel/bzlmod/IndexRegistry.java

```diff
@@ -174,7 +179,10 @@ private Optional<byte[]> doGrabFile(
           checksum = Optional.empty();
         } else {
           // Guarantee reproducibility by assuming that the file still doesn't exist.
-          return Optional.empty();
+          throw new NotFoundException(
+              String.format(
+                  "%s: previously not found (as recorded in %s)",
```

Comment:
Should we also provide instruction on how to renew this by force? E.g. using `--lockfile_mode=refresh`?

Reply:
That's a good idea, done!